h2/src/main/org/h2/mvstore/type/BasicDataType.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2004-2019 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (https://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.mvstore.type;
+
+import org.h2.mvstore.DataUtils;
+import org.h2.mvstore.WriteBuffer;
+import java.nio.ByteBuffer;
+
+/**
+ * Class BasicDataType.
+ * <UL>
+ * <LI> 8/7/19 5:28 PM initial creation
+ * </UL>
+ *
+ * @author <a href='mailto:andrei.tokar@gmail.com'>Andrei Tokar</a>
+ */
+public abstract class BasicDataType<T> implements DataType<T> {
+
+    @Override
+    public abstract int getMemory(T obj);
+
+    @Override
+    public abstract void write(WriteBuffer buff, T obj);
+
+    @Override
+    public abstract T read(ByteBuffer buff);
+
+    @Override
+    public int compare(T a, T b) {
+        if (a == b) {
+            return 0;
+        } else if (a == null) {
+            return -1;
+        } else if (b == null) {
+            return 1;
+        }
+        throw DataUtils.newUnsupportedOperationException("Can not compare");
```

Comment:
What is a purpose of that implementation? Maybe this method should be abstract or throw the exception unconditionally?